src/amoc_api_handler.erl

```diff
@@ -0,0 +1,75 @@
+-module(amoc_api_handler).
+
+-export([init/3]).
+
+-export([rest_init/2,
+         allowed_methods/2,
+         content_types_accepted/2,
+		 content_types_provided/2,
+         from_json/2,
+         process_json/2,
+		 status_response/2]).
+
+-record(state, {action}).
+
+-spec init(tuple(), cowboy:req(), #state{}) -> {upgrade, protocol, cowboy_rest}.
```

Comment:
The spec in line 15 uses a record, please define a type for the record and use that instead.